apps/frontend/src/components/TimeCardPage/SubmitCard.tsx

```diff
@@ -1,61 +1,214 @@
+    if (isSubmitted && statusEntry.Date !== undefined) {
+      setSubmitDate(moment.unix(statusEntry.Date));
+    }
+
+    // Determine the latest status update to set the card state
+    if (props.timesheetStatus.Finalized !== undefined) {
+      setState(CardState.AdminFinalized);
+    } else if (props.timesheetStatus.HoursReviewed !== undefined) {
+      setState(CardState.InReviewAdmin);
+    } else if (props.timesheetStatus.HoursSubmitted !== undefined) {
+      setState(CardState.InReviewSupervisor);
+    } else {
       setState(CardState.Unsubmitted);
     }
-  }
+  }, [currUser, props.timesheetStatus, props.timesheetId]);
+
+  const submitAction = async () => {
+    console.log("Current user id:", currUser.UserID);
+    let statusSubmissionType: string;
+
+    // Determine the appropriate status entry to match up with the logged in user's role
+    switch (currUser.Type) {
+      case UserTypes.Associate:
+        statusSubmissionType = TimesheetStatus.HOURS_SUBMITTED;
+        break;
+
+      case UserTypes.Supervisor:
+        statusSubmissionType = TimesheetStatus.HOURS_REVIEWED;
+        break;
+
+      case UserTypes.Admin:
+        statusSubmissionType = TimesheetStatus.HOURS_SUBMITTED;
+        break;
+    }
+
+    // Update the current timesheet to be submitted by the logged-in user.
+    // The type of status can be determined on the backend by the user type
+    try {
+      const response = await ApiClient.updateTimesheet(
+        updateSchemas.TimesheetUpdateRequest.parse({
+          TimesheetID: props.timesheetId,
+          Operation: updateSchemas.TimesheetOperations.STATUS_CHANGE,
+          Payload: updateSchemas.StatusChangeRequest.parse({
+            TimesheetId: props.timesheetId,
+            AssociateId: props.associateId,
+            authorId: currUser.UserID, // TODO: Implement authorId functionality instead of dummy data
+            statusType: statusSubmissionType,
+            dateSubmitted: moment().unix(),
+          }),
+        })
+      );
+
+      // TODO: Confirm successful 2xx code responSse from API
+      props.refreshTimesheetCallback();
+
+      toast({
+        title: "Successful submission!",
+        status: "success",
+        duration: 3000,
+        isClosable: true,
+      });
+    } catch (err) {
+      console.log(err);
+      toast({
+        title: "Uh oh, something went wrong...",
+        status: "error",
+        duration: 3000,
+        isClosable: true,
+      });
+      return;
+    }
+  };
 
   return (
-    <Box className="col-md-2" style={{ display: "flex", justifyContent: 'flex-end' }}>
+    <Box
+      className="col-md-2"
+      style={{ display: "flex", justifyContent: "flex-end" }}
+    >
       <Card
-        bg={(state === CardState.Completed) ? 'success' : ((state === CardState.InReviewBreaktime || state === CardState.InReviewEmployer) ? 'warning' : 'danger')}
-        textColor="white"
+        bg={
+          state === CardState.AdminFinalized
```

Comment:
can this be extracted into a const?

Reply:
Yeah, just to keep the HTML cleaner you mean?

Reply:
yeah